src/network/mySelfQuota.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { MYSELF_QUOTA_PATH, STORAGES_ENDPOINT } from '../carbonio-files-ui-common/constants';
+
+const getMyQuotaCompleted = (
+	xhr: XMLHttpRequest,
+	resolve: (
+		value: { limit: number; used: number } | PromiseLike<{ limit: number; used: number }>
+	) => void,
+	reject: (reason?: { statusText: string; status: number }) => void
+): void => {
+	switch (xhr.status) {
+		case 200: {
+			const response = JSON.parse(xhr.response);
+			resolve({ limit: response.limit, used: response.used });
+			break;
+		}
+		case 400:
+		case 404:
+		case 413:
+		case 500:
```

Comment:
Considering that all of these fall in the default case, I think we could avoid making them explicit, and use the default directly